src/Components/Shared/SearchBar.tsx

```diff
@@ -0,0 +1,170 @@
+                    <DropdownButton title="Filter" className="filter-dropdown">
+                        <Dropdown.Divider></Dropdown.Divider>
+                        <Form.Check
+                            inline
+                            disabled={!this.props.album}
+                            checked={this.state.types.album}
+                            label="Album"
+                            onChange={() => {
+                                this.setState({ types: { ...this.state.types, album: !this.state.types.album } })
+                            }}
+                        />
+                        <Form.Check
+                            inline
+                            disabled={!this.props.artist}
+                            checked={this.state.types.artist}
+                            label="Artist"
+                            onChange={() => {
+                                this.setState({ types: { ...this.state.types, artist: !this.state.types.artist } })
+                            }}
+                        />
+                        <Form.Check
+                            inline
+                            disabled={!this.props.playlist}
+                            checked={this.state.types.playlist}
+                            label="Playlist"
+                            onChange={() => {
+                                this.setState({ types: { ...this.state.types, playlist: !this.state.types.playlist } })
+                            }}
+                        />
+                        <Form.Check
+                            inline
+                            disabled={!this.props.track}
+                            checked={this.state.types.track}
+                            label="Track"
+                            onChange={() => {
+                                this.setState({ types: { ...this.state.types, track: !this.state.types.track } })
+                            }}
+                        />
+                    </DropdownButton>
+                    <FormControl
+                        as="input"
+                        placeholder="Search"
+                        value={this.state.searchVal}
+                        onKeyDown={this.search}
+                        onChange={(e) => this.setState({ ...this.state, searchVal: e.target.value })}
+                    />
+                </InputGroup>
+                <Accordion>
+                    <Card>
+                        <Accordion.Toggle as={Card.Header} className="search-results-toggle__button" eventKey="0" onSelect={() => {}}>
+                            &#10225;
+                        </Accordion.Toggle>
+                        <Accordion.Collapse eventKey="0">
+                            <ul>{searchItems}</ul>
+                        </Accordion.Collapse>
+                    </Card>
+                </Accordion>
+            </div>
+        )
+    }
+
+    createQuery(): string {
```

Comment:
This should probably be a function in a service. That way it can be tested separate from the component